### PR TITLE
Convert docs to use Jupyter Book

### PR DESCRIPTION
**Pull request by kne42**
# Description
Use Jupyter Book to generate our docs and switch to the Furo theme. Special thanks to @tlambert03 for helping me get set up with everything! :)

Screen Shot 2021-01-29 at 11 15 33 AM

## Building the docs
```bash
make docs
```

and navigate to the link provided on build success.

### CODE REVIEW

**Comment by sofroniewn** [comment]
@kne42 it looks like this is coming along really nicely, but one thing I was wondering is what's happening to all our release notes - it looks like they are being deleted right now. Is that intentional? Is the plan for them to live somewhere else long term?

File: docs/_config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+sphinx:
+  extra_extensions:
+    - sphinx.ext.viewcode
+    - sphinx.ext.napoleon
+    - sphinx.ext.autodoc
+    - sphinx_autodoc_typehints
+    - sphinx.ext.autosummary
+
+  config:
```
Comment (tlambert03) [inline docs/_config.yml:35]: could consider using a different pygments theme... 

I think the default is a bit drab, 

Untitled2

here's `pygments_style: paraiso-dark` ... and you can explore more [here](https://pygments.org/demo/#try).

Untitled

File: docs/_config.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    autosummary_generate: True
+    autosummary_imported_members: True
+    html_theme: furo
+    templates_path:
+        - '_templates'
```
Comment (tlambert03) [inline docs/_config.yml:41]: might be nice to override the footer template and give a shout-out to jupyter-book.  (where it says `Built with Sphinx and @pradyunsg's Furo theme.` ... maybe have it say `Built with Sphinx , Jupyter Book and the Furo theme.` with the appropriate links?)

File: docs/_config.yml (hunk 1 above)
Comment (tlambert03) [inline docs/_config.yml:35]: also, adding an intersphinx mapping to python docs is great for builtin type links:
```
    intersphinx_mapping:
      python:
        - "https://docs.python.org/3"
        - null
```

Comment (tlambert03) [reply docs/_config.yml:35]: woaaaaah.  I just realized that this theme will auto-switch light/dark depending on your system settings.  That's pretty bad-ass (I think?)... so might want to keep that in mind before accepting this suggestion (though, I'm still not a huge fan of the yellowish background)

https://user-images.githubusercontent.com/1609449/106403063-a3710900-63fa-11eb-92bc-e1d317ddfd2d.mov

**Review by tlambert03 — COMMENTED** [review]
Awesome work @kne42.  We're wayyyy better off with this :)
couple thoughts:

- the icon at the top left is a bit huge.
- I think we should remove signatures from the module summary table (for instance, in `_build/html/api/napari.layers.htm` with the `:nosignatures: directive added to module.rst
   ```rst

   .. autosummary::
      :nosignatures:
      :toctree:
   {% for item in classes %}
      {{ item }}
   ```
changes this

to this:


- release notes are missing (I hadn't ported those over yet in my branch).
- this isn't specific to this PR, but we've long had an issue with how sphinx renders our class docs (like `Image`, where the `Attributes` section in the class docs is fighting with the all of the `@property`s).  So there ends up being a _lot_ of duplicated keys and they remain extremely hard to parse.  Could leave that for another PR, but just noting it here.  I think it should be a high priority for cleanup.

- we can now take advantage of myst-markdown goodness... so you might want to do a quick search for places where we used `eval-rst` and modernize them if possible.  for instance, in `rendering.md` change the first table to this to get a proper table:

```md
| Framerate  | Milliseconds  | Experience |
| :--------  | ------------  | ---------: |
| 60Hz       |  16.7         | Great      |
| 30Hz       |  33.3         | Good       |
| 20Hz       |  50           | Acceptable |
| 10Hz       |  100          | Bad        |
| 5Hz        |  200          | Unusable   |
```

File: docs/_config.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+# only_build_toc_files: true
+
+# Force re-execution of notebooks on each build.
+# See https://jupyterbook.org/content/execute.html
+execute:
+  execute_notebooks: force
+
+# Information about where the book exists on the web
+repository:
+  url: https://github.com/executablebooks/jupyter-book # Online location of your book
```
Comment (Carreau) [inline docs/_config.yml]:
```suggestion
  url: https://github.com/napari/napari
```

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+pip install napari[all]
+```
+
+This last option (`pip install napari[all]`) will choose a framework for
+you — currently PyQt5, but this could change in the future.
+
+## tutorials & getting started
+
+If you are just getting started with napari, we suggest you check out the
+tutorials at [napari.org](http://napari.org).  The documentation presented
```
Comment (Carreau) [inline docs/index.md:47]:
```suggestion
tutorials at [napari.org](https://napari.org).  The documentation presented
```

File: docs/_config.yml (hunk 1 above)
Comment (justinelarsen) [reply docs/_config.yml:35]: That's cool! @liaprins-czi I'm not sure if you've thought about this as you're working on the design. It could be good for accessibility too.

Comment (liaprins-czi) [reply docs/_config.yml:35]: Good point, a lot fo apps and websites are allowing to change background themes based on user settings, we could do that too

**Comment by sofroniewn** [comment]
Love how this is coming along @kne42! Let us know when it's ready for re-review or if there is still more stuff you have planned to do!!

**Comment by kne42** [comment]
I believe I've addressed all concerns with the exception of the logo size which is something more design-related, which I'll address when we create our own sphinx theme.

Notes of some things to fix in the future:
- the max depth for the API Reference was set to be one-level deep, so certain modules referenced in documents are not shown. this was done so that e.g. `napari.layers.image.image.Image` would become `napari.layers.Image`. ultimately, I think this has more to do with the structure of our code than the actual documentation, so that should be addressed in another PR
- inherited methods are not showing up in the API reference for some reason, they appear in the table of contents but the actual docstring isn't there
- in the release notes, some links are not working correctly (such as the issue references). I want to address this using [extensions](https://www.sphinx-doc.org/en/master/extdev/index.html) in our sphinx theme

Anyways, this is ready for re-review, so please check it out and let me know if there's anything else that needs to be addressed before merging!

**Comment by tlambert03** [comment]
> I believe I've addressed all concerns with the exception of the logo size which is something more design-related, which I'll address when we create our own sphinx theme.

👍

> ultimately, I think this has more to do with the structure of our code than the actual documentation, so that should be addressed in another PR

sounds good.

> * inherited methods are not showing up in the API reference for some reason, they appear in the table of contents but the actual docstring isn't there

did you try the `:inherited-members:` option for the [autodoc directive](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html)?

> Anyways, this is ready for re-review, so please check it out and let me know if there's anything else that needs to be addressed before merging!

this is a great place to start, thanks @kne42!   will be great to have this in

**Review by tlambert03 — APPROVED** [review]

**Review by sofroniewn — APPROVED** [review]
This looks great @kne42!! Really nice! I'm good to merge now too, will be really exciting to get this in! We've got our 0.4.5 release combining up soon, will be cool to have this for that!
